server/src/authentication.middleware.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Injectable,
+  NestMiddleware,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { AuthenticationService } from './modules/authentication/authentication.service';
+import * as jwt from 'jsonwebtoken';
+
+interface AuthenticatedRequest extends Request {
+  user?: any;
+}
+
+@Injectable()
+export class AuthenticationMiddleware implements NestMiddleware {
```

Comment:
As mentioned on Discord, this should be replaceable with guards & strategies in nestjs/passport: https://docs.nestjs.com/recipes/passport#implement-protected-route-and-jwt-strategy-guards